Makefile

```diff
@@ -27,8 +27,10 @@ IMAGE_VERSION ?= v0.0.21-rc.0
 E2E_IMAGE_VERSION = v0.1.0-e2e-$(BUILD_COMMIT)
 # Use a custom version for E2E tests if we are testing in CI
 ifdef CI
+ifndef RELEASE
```

Comment:
Is `RELEASE` referenced elsewhere here?

Could `e2e-helm-deploy` just point to `E2E_IMAGE_VERSION` instead of doing an override?

Should there be a `helm-deploy` target for non-e2e usecases?

Just trying to think of ways to reduce conditionals here as it can be kind of confusing to follow.

Reply:
> Is RELEASE referenced elsewhere here?

Nope, it's only used for overriding the image version. 

> Could e2e-helm-deploy just point to E2E_IMAGE_VERSION instead of doing an override?

Yeah, as we're renaming the target to explicitly say `e2e-*`, I think it's safe to change the target to use `E2E_IMAGE_VERSION` instead of the override.

> Should there be a helm-deploy target for non-e2e usecases?

Typically users will just install it by adding the chart repo, but when we add the documentation for how to run e2e tests locally, we can add a target for developers who want to install the current driver. WDYT?

Reply:
> Is RELEASE referenced elsewhere here?

Nope, it's only used for overriding the image version. 

> Could e2e-helm-deploy just point to E2E_IMAGE_VERSION instead of doing an override?

The `IMAGE_VERSION` is used in a couple of places other than installing the driver. For instance, image scanning and docker build for the image. We would need to replace all targets. So I think keeping just a single variable that's used across all targets makes it easier for developers trying to build the images locally too. WDYT?

> Should there be a helm-deploy target for non-e2e usecases?

Typically users will just install it by adding the chart repo, but when we add the documentation for how to run e2e tests locally, we can add a target for developers who want to install the current driver. WDYT?

Reply:
I don't understand when `RELEASE` would be set is what is really confusing me here. I think a comment would help explain.

Reply:
Had an offline discussion with Tommy and updated the PR to add a constant which makes it more clear:
```bash
RELEASE_VERSION := v0.0.21
IMAGE_VERSION ?= v0.0.21-rc.0
ifdef CI
override IMAGE_VERSION := v0.1.0-e2e-$(BUILD_COMMIT)
endif
```

Reply:
lgmt